ExperimentA/main.cpp

```diff
@@ -204,38 +204,31 @@ void test(Arguments& arg) {
 		arg.cmcPlotFile.close();
 	}
 
-#pragma omp parallel
+#pragma omp critical
```

Comment:
You can get rid of this (and the corresponding curly braces. The kind of stuff you're doing here is a bad fit for parallelization